skyve-war/src/main/java/modules/admin/Tag/TagService.java

```diff
@@ -0,0 +1,144 @@
+		if (subject != null && object != null) {
+			// no dialect insensitive way to use SQL for creating new UUIDs for new records
+			// add tagged items from object tag to subject tag
+			// TagManager function deals with duplicates
+			TagManager tm = EXT.getTagManager();
+			try (AutoClosingIterable<Bean> i = tm.iterate(object.getBizId())) {
+				for (Bean bean : i) {
+					tm.tag(subject.getBizId(), bean);
+				}
+			}
+			subject.setUploadTagged(
+					Long.valueOf(subject.countDocument(subject.getUploadModuleName(), subject.getUploadDocumentName())));
+			subject.setTotalTagged(Long.valueOf(subject.count()));
+		}
+	}
+
+	/**
+	 * Deletes from the subject Tag any items which are not also in the object
+	 * Tag.
+	 * 
+	 * @param subject
+	 * @param object
+	 * @throws Exception
+	 */
+	@SuppressWarnings("static-method")
+	public void intersect(TagExtension subject, TagExtension object) throws Exception {
+		if (subject != null && object != null) {
+			Persistence pers = CORE.getPersistence();
+
+			// unsecured SQL for performance
+			StringBuilder intersect = new StringBuilder();
+			intersect.append("delete from ADM_Tagged ");
+			intersect.append(" where taggedBizId not in (");
+			intersect.append(" select taggedBizId from ADM_Tagged where ");
+			intersect.append(" tag_id = :objectTagId");
+			intersect.append(" and bizCustomer=:objectBizCustomer").append(" ) ");
+			intersect.append(" and tag_id =:subjectTagId");
+			intersect.append(" and bizCustomer=:subjectBizCustomer");
+
+			SQL sql = pers.newSQL(intersect.toString());
+			sql.putParameter("objectTagId", object.getBizId(), false);
+			sql.putParameter("subjectTagId", subject.getBizId(), false);
+			sql.putParameter("objectBizCustomer", subject.getBizCustomer(), false);
+			sql.putParameter("subjectBizCustomer", subject.getBizCustomer(), false);
+
+			sql.execute();
+
+			subject.setUploadTagged(
+					Long.valueOf(subject.countDocument(subject.getUploadModuleName(), subject.getUploadDocumentName())));
+			subject.setTotalTagged(Long.valueOf(subject.count()));
+		}
+	}
+
+	/**
+	 * Deletes from the subject Tag items which are in the object Tag.
+	 * 
+	 * @param subject
+	 * @param object
+	 * @throws Exception
+	 */
+	@SuppressWarnings("static-method")
+	public void except(TagExtension subject, TagExtension object) throws Exception {
+		if (subject != null && object != null) {
+			TagManager tm = EXT.getTagManager();
+			try (AutoClosingIterable<Bean> i = tm.iterate(object.getBizId())) {
+				for (Bean bean : i) {
+					// TagManager method handles if this bean was not tagged
+					tm.untag(subject.getBizId(), bean);
+				}
+			}
+			subject.setUploadTagged(
+					Long.valueOf(subject.countDocument(subject.getUploadModuleName(), subject.getUploadDocumentName())));
+			subject.setTotalTagged(Long.valueOf(subject.count()));
+		}
+	}
+
+	/**
+	 * Retrieve the items tagged which match the specified module and document
+	 * 
+	 * @param mailout
+	 * @return
+	 * @throws Exception
```

Comment:
Clean up Javadoc when we touch it. Parameters should have descriptions for developers using the service so they know what is expected. So should return types.